vulnerabilities/pipelines/collect_commits.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# VulnerableCode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/aboutcode-org/vulnerablecode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+from aboutcode.pipeline import LoopProgress
+
+from vulnerabilities.models import CodeFix
+from vulnerabilities.models import Package
+from vulnerabilities.models import VulnerabilityReference
+from vulnerabilities.pipelines import VulnerableCodePipeline
+from vulnerabilities.utils import normalize_purl
+
+
+class CollectFixCommitsPipeline(VulnerableCodePipeline):
+    """
+    Improver pipeline to scout References and create CodeFix entries.
+    """
+
+    pipeline_id = "collect_fix_commits"
+    license_expression = None
+
+    @classmethod
+    def steps(cls):
+        return (cls.collect_and_store_fix_commits,)
+
+    def collect_and_store_fix_commits(self):
+        references = VulnerabilityReference.objects.prefetch_related("vulnerabilities").distinct()
+
+        self.log(f"Processing {references.count():,d} references to collect fix commits.")
+
+        created_fix_count = 0
+        progress = LoopProgress(total_iterations=references.count(), logger=self.log)
+        for reference in progress.iter(references.paginated(per_page=500)):
+            for vulnerability in reference.vulnerabilities.all():
+                package_urls = self.extract_package_urls(reference)
+                commit_id = self.extract_commit_id(reference.url)
+
+                if commit_id and package_urls:
+                    for purl in package_urls:
+                        normalized_purl = normalize_purl(purl)
+                        package = self.get_or_create_package(normalized_purl)
```

Comment:
Do we really want to create new packages here? I would have thought we are only adding new CodeFix?